src/platform_impl/linux/wayland/event_loop/mod.rs

```diff
@@ -68,6 +71,56 @@ pub struct EventLoop {
     // XXX drop after everything else, just to be safe.
     /// Calloop's event loop.
     event_loop: calloop::EventLoop<'static, WinitState>,
+
+    pump_event_notifier: Option<PumpEventNotifier>,
+}
+
+#[derive(Debug)]
+struct PumpEventNotifier {
+    /// Whether we're in winit or not.
+    control: Arc<(Mutex<bool>, Condvar)>,
+    /// Thread handle.
+    _handle: JoinHandle<()>,
+}
+
+fn spawn_wakeup_thread(
+    connection: Connection,
+    awakener: calloop::ping::Ping,
+) -> Option<PumpEventNotifier> {
+    // Start from the waiting state.
+    let control = Arc::new((Mutex::new(true), Condvar::new()));
+    let control_thread = Arc::clone(&control);
+
+    let handle = std::thread::Builder::new()
+        .name(String::from("wayland pump_events wake-up notify"))
```

Comment:
The first 15 character should contain enough information to identify the thread since everything after that won't be visible in non-rust APIs.